### PR TITLE
Add progress bar for each module 

### PR DESCRIPTION
**Pull request by tlx02**
Updated the GUI to display a progress bar for each module. 
The progress bar indicates the percentage of completed tasks for the module.

### CODE REVIEW

File: src/main/java/seedu/address/model/module/DistinctModuleList.java [hunk 1]
```diff
@@ -52,6 +53,46 @@ public void setModules(List<Module> modules) {
...
+    /**
+     * Counts the number of tasks in {@code tasks} that belong to {@code module},
+     * and updates this number in {@code module}.
+     * {@code module} must exist in the module list.
+     *
+     * @param module The module to check for number of tasks.
+     * @param tasks the list of tasks to check with the module.
+     */
+    public void updateTotalNumOfTasks(Module module, DistinctTaskList tasks) {
+        int totalNumOfTasks = tasks.getTotalNumOfTasks(module);
```
Comment (dlimyy) [inline src/main/java/seedu/address/model/module/DistinctModuleList.java:65]: Perhaps you could look into including null checks for both module and DistinctTaskList here(More about defensive programming)

**Review by dlimyy — APPROVED** [review]
Overall, the code LGTM!

**Comment by dlimyy** [comment]
There is a merge conflict here that has to be resolved.

File: src/main/java/seedu/address/model/module/DistinctModuleList.java (hunk 1 above)
Comment (tlx02) [reply src/main/java/seedu/address/model/module/DistinctModuleList.java:66]: I agree with you, I have included the null checks.

**Pull request merged**